### DISCUSSION TITLE
Fake GC-01 (CA32F0502L3): Tube voltage

### DISCUSSION THREAD

**Discussion by boromix** [Gissio/radpro#114] [Q&A] [answered]
Thanks Gissio for v2.01.  J321 tube data sheet reports recommendation work voltage 390V but measuring j321 voltage on my Fnirsi GC-01 I read 712V using 1 Gohm probe.      https://www.theremino.com/wp-content/uploads/2013/03/Sensor_HvProbe.zip
That's weird because on Bosean FS-600 is 400V as espected. Any Idea?

**Comment by boromix** [comment, marked as answer]
Solved. removed R38 and put 10.68 Mohm. Now 402V at J321 tube

https://www.dropbox.com/scl/fi/c5epoyxuq8ehgfekaciri/FNIRSI-2024-06-25_003.jpeg?rlkey=j4anwttxj29zivwcmxqujpaf1&st=g1b0ypix&dl=0

https://www.dropbox.com/scl/fi/u9nta2fyf0t6aobu4nzjw/FNIRSI-2024-06-25-002.jpeg?rlkey=ksix03jo4oxn4fzdxs1xzo4d5&st=1tanntx4&dl=0

**Reply by paulc317** [reply]
Dropbox links aren't valid anymore. 
Please update them.

**Comment by gc01noob** [comment]
by any chance do you have a part number, supplier or listing for a 10.68M resistor?

I'm yet to test mine but would be good to know if required.

**Reply by boromix** [reply]
here the 712V measured with 1Gohm probe on J321 tube as arrive from aliex

https://www.dropbox.com/scl/fi/xsdsam25l77mo8ytnln6e/FNIRSI-2024-06-25_004.jpeg?rlkey=abqboku6iv19wq4hicmova8aj&st=aqjixgy6&dl=0

I removed R38, put different resistors in series until I obtained the value of 400V on my geiger than I soldered a 1/8W 10Mohm resistor plus a 680Kohm SMD 0805 as you can see in picture

https://www.dropbox.com/scl/fi/u9nta2fyf0t6aobu4nzjw/FNIRSI-2024-06-25-002.jpeg?rlkey=ksix03jo4oxn4fzdxs1xzo4d5&st=1zf27qzq&dl=0

fast and not bad reworking

**Reply by ioderey** [reply]
I have 630V default on FNIRSI GC-01 (CH32F103R8T6) J321. I set custom HV friqency 5000 and 0.30 duty cycle, HV is 405V now.

**Comment by aenglebert** [comment]
Hi @boromix.

My message is a bit out of scope for this closed issue, I'm sorry for that, but I can't find any other way to contact you (no email or other information on your github profile).

I see that you mentioned CA32F0502L3 in the title of this issue, on the radpro github.
Is it possible that you were able to flash this MCU with radpro?
If so, could you please share the parameters you used?
Like using one of the existing configurations for the APM32 or CH32, or by creating a custom json board file/gc01.h file/... ?
  
My only other lead at the moment is the name (in Chinese) of a factory that seems to produce the chip, but doesn't provide more information publicly.

Thanks, and sorry again for being out of scope, we can move the conversation elsewhere if needed.

**Reply by KountKartoffel** [reply]
@boromix @aenglebert I would be very interested as well as I have a CACHIP MCU as well.

**Reply by Gissio** [reply]
Let me cite from Rad Pro's documentation:

> **WARNING:** Rad Pro is incompatible with the FNIRSI GC-01 with a CACHIP microprocessor. Installing Rad Pro on such a device WILL BREAK IT.

**Reply by aenglebert** [reply]
Hi @Gissio.
It is very clear to me that the CACHIP is not supported and I fully understand that with the lack of documentation regarding this chip, no official support will be put in place.
This is not my request (having maintained an alternative Android OS for a tablet, I know how tiring repeated requests for support for unrelated things can be).
I'm just exploring the possibility of compiling an adapted version myself, and since I saw this post with a reference to a CA32F0502L3 device, I asked if anyone had found some interesting parameters.
This CACHIP is probably a cheap STM32 clone, so it should be technically possible to compile a version, the main obstacle being that we are in the dark about the details of this chip...
But you're right that it's worth remembering for anyone reading this that it's not safe and will break the device to just flash it with the existing version.

**Reply by Gissio** [reply]
aenglebert: Thanks for your message and for your understanding ("I know how tiring repeated requests [...] can be"). It's always good to put oneself in the other person's shoes.

If you have the time and motivation, this is what would need to be done: https://github.com/Gissio/radpro/issues/132#issuecomment-2303647221

And as you mentioned, I just wrote this warning to prevent anyone from attempting something that **WILL** break things!
